### PR TITLE
Create main.yml

### PR DESCRIPTION
**Pull request by superboyiii**
Create main.yml

### CODE REVIEW

**Comment by shargon** [comment]
I requires a `dotnet format`

**Comment by superboyiii** [comment]
> It requires a `dotnet format`

Do you mean fix these?
```
2020-07-14T09:01:35.3771534Z   StatesDumper/StatesDumper.cs(145,1): Fix whitespace formatting.
2020-07-14T09:01:35.4241064Z   RpcNep5Tracker/Nep5BalanceKey.cs(65,29): Fix whitespace formatting.
2020-07-14T09:01:35.4241683Z   RpcNep5Tracker/Nep5BalanceKey.cs(66,29): Fix whitespace formatting.
2020-07-14T09:01:35.4242099Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(257,41): Fix whitespace formatting.
2020-07-14T09:01:35.4248972Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(257,51): Fix whitespace formatting.
2020-07-14T09:01:35.4249434Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(260,45): Fix whitespace formatting.
2020-07-14T09:01:35.4249787Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(260,55): Fix whitespace formatting.
2020-07-14T09:01:35.4250165Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(261,43): Fix whitespace formatting.
2020-07-14T09:01:35.4250509Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(261,53): Fix whitespace formatting.
2020-07-14T09:01:35.4250851Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(262,45): Fix whitespace formatting.
2020-07-14T09:01:35.4251284Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(262,55): Fix whitespace formatting.
2020-07-14T09:01:35.4251646Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(287,36): Fix whitespace formatting.
2020-07-14T09:01:35.4251979Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(287,48): Fix whitespace formatting.
2020-07-14T09:01:35.4252326Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(293,41): Fix whitespace formatting.
2020-07-14T09:01:35.4252669Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(293,51): Fix whitespace formatting.
2020-07-14T09:01:35.4253006Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(311,36): Fix whitespace formatting.
2020-07-14T09:01:35.4253336Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(311,48): Fix whitespace formatting.
2020-07-14T09:01:35.4253671Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(342,41): Fix whitespace formatting.
2020-07-14T09:01:35.4254004Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(342,51): Fix whitespace formatting.
2020-07-14T09:01:35.4254344Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(343,43): Fix whitespace formatting.
2020-07-14T09:01:35.4254671Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(343,53): Fix whitespace formatting.
2020-07-14T09:01:35.4254999Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(344,41): Fix whitespace formatting.
2020-07-14T09:01:35.4255352Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(344,51): Fix whitespace formatting.
2020-07-14T09:01:35.4255679Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(357,41): Fix whitespace formatting.
2020-07-14T09:01:35.4256008Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(357,51): Fix whitespace formatting.
2020-07-14T09:01:35.4265341Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(383,42): Fix whitespace formatting.
2020-07-14T09:01:35.4265784Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(383,54): Fix whitespace formatting.
2020-07-14T09:01:35.4266134Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(384,41): Fix whitespace formatting.
2020-07-14T09:01:35.4266482Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(384,101): Fix whitespace formatting.
2020-07-14T09:01:35.4266818Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(392,36): Fix whitespace formatting.
2020-07-14T09:01:35.4267172Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(406,57): Fix whitespace formatting.
2020-07-14T09:01:35.4267784Z   RpcSystemAssetTracker/RpcSystemAssetTrackerPlugin.cs(406,67): Fix whitespace formatting.
```

**Comment by shargon** [comment]
Yes, but I think that this will produce a lot of changes, for 2x, i think that we can avoid this check

**Comment by shargon** [comment]
I am not sure why coveralls doesn't work

**Comment by superboyiii** [comment]
> I am not sure why coveralls doesn't work

Strange....

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      with:
+        dotnet-version: ${{ env.DOTNET_VERSION }}
+    - name: Check format
+      run: |
+        dotnet tool install --version 3.2.111002 --tool-path ./ dotnet-format --add-source https://dotnet.myget.org/F/format/api/v3/index.json
+        # ./dotnet-format --check --dry-run -v diagnostic
+    - name: Test
+      if: runner.os == 'Windows'
+      run: |
+        forfiles /p tests /m *.csproj /s /c "cmd /c dotnet add @PATH package coverlet.msbuild"
```
Comment (cloud8little) [inline .github/workflows/main.yml:32]: remove /p tests; there is no tests folder under current directory.

**Comment by shargon** [comment]
We can fix coveralls later, with IT it's enough

File: .github/workflows/main.yml (hunk 1 above)
Comment (shargon) [reply .github/workflows/main.yml:32]: @cloud8little sorry i didn't read your review, could you fix it in other pr?

**Pull request merged**